assets/src/scripts/inc/history.js

```diff
@@ -0,0 +1,282 @@
+var diff_match_patch = require("exports?diff_match_patch!./../vendor/diff_match_patch.js");
+
+/**
+ * Diff component.
+ */
+var PageHistoryDiff = React.createClass({
+
+	getDefaultProps: function() {
+	    return {
+	    	diff_a: null,
+	    	diff_b: null,
+	    }
+	},
+
+	render: function() {
+		return ( <div className="PageHistory_Diff" dangerouslySetInnerHTML={ { __html: this.getDiffHTML() } } /> );
+	},
+
+	getDiffHTML: function() {
+
+		if ( ! ( this.props.diff_a && this.props.diff_b ) ) {
+			return '';
+		}
+
+		var html = [];
+		var dmp  = new diff_match_patch();
+
+		var diff = dmp.diff_main(
+			this.props.diff_b.content.replace( /[\n|\r]{2,}/g, "\n\n" ),
+			this.props.diff_a.content.replace( /[\n|\r]{2,}/g, "\n\n" )
+		);
+
+		dmp.diff_cleanupSemantic( diff );
+
+		diff.forEach( function( part, i ) {
+
+			var text = this.escapeHtml( part[1] );
+
+			if ( 1 === part[0] ) {
+				html[i] = '<ins class="PageHistory_Diff-Added">' + text + '</ins>';
+			} else if ( -1 === part[0] ) {
+				html[i] = '<del class="PageHistory_Diff-Removed">' + text + '</del>';
+			} else {
+				html[i] = '<span>' + text + '</span>';
+			}
+
+		}.bind( this ) );
+
+		return html.join( '' );
+
+	},
+
+	escapeHtml: function ( text ) {
+
+		var map = {
+			'&': '&amp;',
+			'<': '&lt;',
+			'>': '&gt;',
+			'"': '&quot;',
+			"'": '&#039;',
+			"/\r|\n": '&para;',
+		};
+
+		return text.replace( /[&<>"']/g, function( m ) { return map[ m ]; } );
+
+	},
+
+});
+
+/**
+ * History List.
+ */
+var PageHistoryList = React.createClass({
+
+	getDefaultProps: function() {
+		return {
+			revisions: [],
+		};
+	},
+
+	getInitialState: function() {
+		return {
+			revisions: this.props.revisions,
+		};
+	},
+
+	render: function() {
+
+		const actions = {
+			selectRevision: this.props.actions.selectRevision,
+		};
+
+		return (
+			<div className="PageHistory_List_Container">
+				<h4 className="PageHistory_List_Title">Page History</h4>
+				<ul className="PageHistory_List">
+					{ this.state.revisions.map( function( revision ) {
+
+						if ( ! 'active' in revision ) {
+							revision.active = false;
+						}
+
+						return (
+							<PageHistoryListItem key={ revision.id } id={ revision.id } action={ revision.action } author={ revision.author } date={ revision.date } content={ revision.content } active={ revision.active } actions={ actions } />
+						);
+					}) }
+				</ul>
+			</div>
+		);
+
+	},
+
+});
+
+var PageHistoryListItem = React.createClass({
+
+	getDefaultProps: function() {
+		return {
+			id:      0,
+			active:  false,
+			author:  '',
+			action:  'update',
+			date:    '',
+			content: '',
+		};
+	},
+
+	render: function() {
+
+		var verb       = 'create' === this.props.action ? 'created' : 'updated';
+		var classNames = [ 'PageHistory_List_Item' ];
+
+		if ( this.props.active ) {
+			classNames.push( 'PageHistory_List_Item-Active' );
+		}
+
+		return (
+			<li className={ classNames.join( ' ' ) } onClick={ this.handleSelectRevision }>
```

Comment:
Maybe use an anonymous for the click handler? `e => this.props.actions.selectRevision( this.props )`

Generally, action callbacks are called `on...` rather than being under a subkey as well.